servicetalk-encoding-api/src/main/java/io/servicetalk/encoding/api/IdentityContentCodec.java

```diff
@@ -62,4 +61,19 @@ public Buffer decode(final Buffer src, final int offset, final int length, final
     public Publisher<Buffer> decode(final Publisher<Buffer> from, final BufferAllocator allocator) {
         return from;
     }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other instanceof ContentCodec) {
+            ContentCodec contentCodec = (ContentCodec) other;
+            return hashCode() == contentCodec.hashCode() && CharSequences.contentEquals(name(), contentCodec.name());
```

Comment:
As you already figured out, we can not guarantee that the developer who extends this interface wont break contract by following a different equality strategy. Its better to keep our strategy simple and only match on name.
Two codecs with equal name, but not equal hashcode smell problem and its good if we can reject these when we check for equality in the various call-sites where we check for `identity`.